trl/trainer/rloo_trainer.py

```diff
@@ -0,0 +1,700 @@
+import gc
+import os
+import time
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Dict, List, Literal, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from accelerate import Accelerator
+from accelerate.state import AcceleratorState
+from accelerate.utils import broadcast, gather_object
+from datasets import Dataset
+from torch.utils.data import DataLoader
+from transformers import (
+    DataCollatorWithPadding,
+    GenerationConfig,
+    PreTrainedTokenizer,
+    Trainer,
+    TrainerCallback,
+    TrainerControl,
+    TrainerState,
+    TrainingArguments,
+)
+from transformers.integrations import get_reporting_integration_callbacks
+from transformers.trainer_callback import CallbackHandler, DefaultFlowCallback
+
+from trl.models.utils import unwrap_model_for_generation
+from trl.trainer.utils import exact_div, print_rich_table
+
+
+INVALID_LOGPROB = 1.0
+
+
+@dataclass
+class RLOOConfig(TrainingArguments):
+    # common config
+    exp_name: str = os.path.basename(__file__)[: -len(".py")]
+    """the name of this experiment"""
+    run_name: Optional[str] = None
+    """a unique name of this run"""
+    sanity_check: bool = False
+    """wether to run in debug mode"""
+
+    # various batch sizes
+    world_size: Optional[int] = None
```

Comment:
Same comment here about hiding this from the user and setting in the trainer itself